src/components/Button/Button.tsx

```diff
@@ -205,9 +211,71 @@ export function Button({
     );
 
   const type = submit ? 'submit' : 'button';
+  const ariaPressedStatus = pressed !== undefined ? pressed : ariaPressed;
+
+  const [disclosureActive, setDisclosureActive] = useState(false);
+  const toggleDisclosureActive = useCallback(() => {
+    setDisclosureActive((disclosureActive) => !disclosureActive);
+  }, []);
+
+  let connectedDisclosureMarkup;
+
+  if (connectedDisclosure) {
+    const connectedDisclosureClassName = classNames(
+      styles.Button,
+      primary && styles.primary,
+      outline && styles.outline,
+      size && size !== DEFAULT_SIZE && styles[variationName('size', size)],
+      textAlign && styles[variationName('textAlign', textAlign)],
+      destructive && styles.destructive,
+      connectedDisclosure.disabled && styles.disabled,
+      styles.iconOnly,
+      styles.ConnectedDisclosure,
+    );
+
+    const defaultLabel = i18n.translate(
+      'Polaris.Button.connectedDisclosureAccessibilityLabel',
+    );
+
+    const {
+      disabled,
+      accessibilityLabel: disclosureLabel = defaultLabel,
+    } = connectedDisclosure;
+
+    const connectedDisclosureActivator = (
+      <button
+        type={type}
```

Comment:
I think we might be safer by typing this as button vs passing the prop.

Reply:
💯